src/package/packageVersionList.ts

```diff
@@ -28,6 +28,13 @@ const VERBOSE_SELECT =
   'ConvertedFromVersionId, Package2.IsOrgDependent, ReleaseVersion, BuildDurationInSeconds, HasMetadataRemoved, CreatedById ' +
   'FROM Package2Version';
 
+const VERBOSE_SELECT_57 =
```

Comment:
@sethmuss is there any other way the new field can be isolated other than creating a query specific for that field?

Reply:
We can use arrays of fields to build the queries to keep it DRY.  E.g., 
```
const defaultFields = ['Id', 'Package2Id', ...];
const verboseFields = ['CodeCoverage', ...];
const verbose57Fields = ['Language'];
let queryFields = defaultFields;
if (options.verbose) {
  queryFields = [...queryFields, ...verboseFields];
  if (connection.version >= 57) {
    queryFields = [...queryFields, ...verbose57Fields];
  }
}
const query = `SELECT ${queryFields.toString()} FROM Package2Version`;
```

The only other way I can think to handle this API version specific code is to assume the field exists, wrap the query in a try/catch and if it fails display a warning message that they can't use the Language feature unless they update to v57.0+, then send the more basic verbose query.

Reply:
But we really don't need all that right? As soon as App server 242 rolls out we'll move the current  VERBOSE_SELECT_57 to VERBOSE_SELECT. I think logical DRY: great. But it's just an extra static field.

I'll admit part of the reason I've written it the way I have is I'm not confident with javascript :)  Like: 
"queryFields = [...queryFields, ...verboseFields];" no idea what those "..."'s are nor the rules around literal arrays(?), and what the gotchas might be. I think if the packaging team is going to own this stuff either: we need to invest in becoming javascript experts, or, keep things as simple as possible.

Reply:
@sethmuss Sorry for the delay getting back to you.

 IIRC, when the field was added it carried the attribute of which version of app server the field was introduced. So FLS is going to reject requests using this field if a user uses an API call with a version < 242. 

Having to account for API version backwards compatibility in this way is not optimal.

I have to admit that version handling for FLS at the sobject level in CLI libraries and even jsforce are almost nonexistent, probably for the same reason that started this discussion.

These are the options:
* Like Steve's example, isolate the knowledge of the new field being version sensitive and use it only when api version is suitable
* try/catch - examine the exception for the appropriate message, adjust the request and retry
* Fetch the sobject describe. The results will only include fields that one has access to, so the request can then be tailored to those fields (jsforce already supports this, so getting it to work is not difficult).
* Use the pattern in the PR
* Wait to merge this PR into main until 242 R2. We have the capability to produce prerelease version of the plugin so your team can do the needed testing before R2.

**Ultimately this will be the packaging teams decision, because it will be yours' to maintain.**

Yes, the team will need to get additional training on javascript and typescript.

We are happy to pair with you and other team members if you need help.